include/signal.h

```diff
@@ -0,0 +1,50 @@
+#ifndef _SYS_SIGNAL_H_
+#define _SYS_SIGNAL_H_
+
+#include <common.h>
+#include <bitstring.h>
+#include <queue.h>
+#include <mutex.h>
+#include <signum.h>
+
+typedef struct proc proc_t;
+
+typedef bitstr_t sigset_t[bitstr_size(SIG_LAST)];
+
+typedef void (*sighandler_t)(int);
+
+#define SIG_DFL (sighandler_t *)0x00
+#define SIG_IGN (sighandler_t *)0x01
+#define SIG_TERM (sighandler_t *)0x02
```

Comment:
I'd prefer to name it `SIGACT_TERM` which wouldn't be as ambiguous, but it must match the naming convention of `SIG_DFL` and `SIG_IGN`, and I cannot rename these two to `SIGACT_*` as these values are also used by userspace programs and their names there must stay `SIG_*`, and we are very strict about not defining the same value twice.

*Side note: I am unhappy with the current forceful merging of user and kernel headers, it implicitly mixes up usermode and kernelmode code, but I'll discuss a solution for this sometime later.*